### PR TITLE
memory game changeCardStatePropToOpposite function

### PR DESCRIPTION
**Pull request by tmansuroglu**
#22

### CODE REVIEW

File: src/components/memorygame/changeCardStatePropertyToOpposite.jsx
Comment (louisrli) [inline src/components/memorygame/changeCardStatePropertyToOpposite.jsx]: you dont want to put functions one per file like this. memorygame/gameHelpers.jsx

File: src/components/memorygame/changeCardStatePropertyToOpposite.jsx
Comment (louisrli) [inline src/components/memorygame/changeCardStatePropertyToOpposite.jsx]: same comment here as your other prs

File: src/components/memorygame/changeCardStatePropertyToOpposite.jsx
Comment (louisrli) [inline src/components/memorygame/changeCardStatePropertyToOpposite.jsx]: using line numbers in comments guarantees that your comment will be useless when someone else that's not you edits this file, unless you are expecting every teammate to read every comment and look for things like this

File: src/components/memorygame/changeCardStatePropertyToOpposite.jsx
Comment (majdajroudi) [inline src/components/memorygame/changeCardStatePropertyToOpposite.jsx]: Where is the property prop coming from? I got a bit confused with the if conditions

File: src/components/memorygame/changeCardStatePropertyToOpposite.jsx
Comment (majdajroudi) [inline src/components/memorygame/changeCardStatePropertyToOpposite.jsx]: is this fn called on click of the card?

**Review by majdajroudi — COMMENTED** [review]
generally readable but needs more comments at the top to explain where the function is called

File: src/components/memorygame/changeCardStatePropertyToOpposite.jsx
Comment (tmansuroglu) [reply src/components/memorygame/changeCardStatePropertyToOpposite.jsx]: Since this is a function, property is not a prop. It is a parameter. Name is misleading but function changes property of a state.So, property stands for target state property. I added a comment about where to see the use of function. Also, I have changed property into stateProperty. I'm not sure if it is descriptive enough. Please let me know

PTAL

File: src/components/memorygame/changeCardStatePropertyToOpposite.jsx
Comment (tmansuroglu) [reply src/components/memorygame/changeCardStatePropertyToOpposite.jsx]: PTAL

File: src/components/memorygame/changeCardStatePropertyToOpposite.jsx
Comment (tmansuroglu) [reply src/components/memorygame/changeCardStatePropertyToOpposite.jsx]: PTAL

File: src/components/memorygame/changeCardStatePropertyToOpposite.jsx
Comment (tmansuroglu) [reply src/components/memorygame/changeCardStatePropertyToOpposite.jsx]: PTAL

File: src/components/memorygame/changeCardStatePropertyToOpposite.jsx
Comment (tmansuroglu) [reply src/components/memorygame/changeCardStatePropertyToOpposite.jsx]: added comment PTAL

File: src/components/memorygame/gameHelpers.jsx
Comment (louisrli) [inline src/components/memorygame/gameHelpers.jsx]: invertCardStateForKey 

we call them keys usually. bc objects are key value pairs

File: src/components/memorygame/gameHelpers.jsx
Comment (louisrli) [inline src/components/memorygame/gameHelpers.jsx]: cardstatekey

File: src/components/memorygame/gameHelpers.jsx
Comment (louisrli) [inline src/components/memorygame/gameHelpers.jsx]: don't put comments above else like this. it's confusing because it's in the if condition above it. put it below the else

Comment (louisrli) [reply src/components/memorygame/gameHelpers.jsx]: you can just delete this comment I think. no need to describe isflipped here, islfipped is probably the most intuitive variable in the code

File: src/components/memorygame/gameHelpers.jsx
Comment (louisrli) [inline src/components/memorygame/gameHelpers.jsx]: uhh aren't your teammates supposed to be using the same shuffle? what happened to that

**Review by louisrli — CHANGES_REQUESTED** [review]

File: src/components/memorygame/gameHelpers.jsx
Comment (tmansuroglu) [reply src/components/memorygame/gameHelpers.jsx]: makes sense PTAL

File: src/components/memorygame/gameHelpers.jsx
Comment (tmansuroglu) [reply src/components/memorygame/gameHelpers.jsx]: PTAL

File: src/components/memorygame/gameHelpers.jsx
Comment (tmansuroglu) [reply src/components/memorygame/gameHelpers.jsx]: comment deleted PTAL

File: src/components/memorygame/gameHelpers.jsx
Comment (tmansuroglu) [reply src/components/memorygame/gameHelpers.jsx]: Correct. shuffle moved back to its previous file. PTAL

File: src/components/memorygame/gameHelpers.jsx
Comment (majdajroudi) [inline src/components/memorygame/gameHelpers.jsx]: you should change this comment to `cardStateKey example: ...`

File: src/components/memorygame/gameHelpers.jsx
Comment (majdajroudi) [inline src/components/memorygame/gameHelpers.jsx]: `invertCardStateForKey is called ...`

File: src/components/memorygame/gameHelpers.jsx
Comment (majdajroudi) [inline src/components/memorygame/gameHelpers.jsx]: `... passed to invertCardStateForKey`

**Review by majdajroudi — CHANGES_REQUESTED** [review]

**Comment by majdajroudi** [comment]
Generally good but comments need to be updated

File: src/components/memorygame/gameHelpers.jsx
Comment (louisrli) [reply src/components/memorygame/gameHelpers.jsx]: majd has some really good points here in general tarkan, which for both of you, the general principle will be a good takeaway. it's good that both of you are writing more comments and developing an intuition for whats readable. the next step is also to write "maintainable" comments

even a small change requested can cause your comments be outdated. and we (we as in we programmers) dont have time to remember comment dependencies (even in a typed language like typescript, this is not "typed" properly in comments, so your code will still work even if comments are outdated)

the lesson / goal here is to write your comments to be as futureproof as possible (its not completely possible and you dont want to be vague but its a goal to aim for). for example, avoiding line numbers. or, when you're naming props, you can put the comment above the prop for example without mentioniing its name (in JS theres not a good way to do this). thus even if you rename the prop the comment is still above the prop

**Review by louisrli — COMMENTED** [review]
when majd approves i will approve

File: src/components/memorygame/gameHelpers.jsx
Comment (tmansuroglu) [reply src/components/memorygame/gameHelpers.jsx]: By saying `you can put the comment above the prop for example without mentioniing its name` are you referring to something like below? @louisrli 
```
function(
  // does bla bla
  param1,
  // does different bla
  param2
)
```

File: src/components/memorygame/gameHelpers.jsx
Comment (tmansuroglu) [reply src/components/memorygame/gameHelpers.jsx]: PTAL

File: src/components/memorygame/gameHelpers.jsx
Comment (tmansuroglu) [reply src/components/memorygame/gameHelpers.jsx]: PTAL

**Review by majdajroudi — APPROVED** [review]

**Comment by louisrli** [comment]
yes

**Comment by louisrli** [comment]
to the comments thing

**Comment by tmansuroglu** [comment]
@louisrli majd approved it

**Comment by tmansuroglu** [comment]
@louisrli waiting for approval

**Review by louisrli — APPROVED** [review]

**Comment by tmansuroglu** [comment]
@Ammar-64 ready for merge

**Pull request merged**
